domino_generator/lib/src/component_generator.dart

```diff
@@ -166,6 +167,37 @@ class ComponentGenerator {
     }
   }
 
+  void _renderText(Stack stack, XmlText node) {
+    final intlAlias = _importAlias('package:intl/intl.dart', ['Intl']);
+    // TODO: deterministic name without causing conflicts
+    final fn_name = 'text_' + Random().nextInt(99999).toString();
```

Comment:
How about:
- first few words limited to 8-16 characters, plus
- SHA hash (of the entire text) limited to 6-12 characters

Reply:
Done. Each text rendering call will have a unique scope to avoid confilicts.